openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1340,6 +1340,7 @@ This specification defines the following Credential Issuer Metadata:
   * `enc_values_supported`: REQUIRED. Array containing a list of the JWE [@!RFC7516] encryption algorithms (`enc` values) [@!RFC7518] supported by the Credential and Batch Credential Endpoint to encode the Credential or Batch Credential Response in a JWT [@!RFC7519].
   * `encryption_required`: REQUIRED. Boolean value specifying whether the Credential Issuer requires the additional encryption on top of TLS for the Credential Response. If the value is `true`, the Credential Issuer requires the encryption for every Credential Response and therefore the Wallet MUST provide encryption keys in the Credential Request. If the value is `false`, the Wallet MAY chose whether it provides encryption keys or not.
 * `credential_identifiers_supported`: OPTIONAL. Boolean value specifying whether the Credential Issuer supports returning `credential_identifiers` parameter in the `authorization_details` Token Response parameter, with `true` indicating support. If omitted, the default value is `false`.
+* `signed_metadata`: OPTIONAL. String that is an entire signed JWT. This JWT can contain Credential Issuer metadata parameters as claims. The signed metadata MUST be secured using JSON Web Signature (JWS) [@!RFC7515] and MUST contain an `iat` (Issued At) claim and an `iss` (Issuer) claim denoting the party attesting to the claims in the signed metadata. This JWT can also be used to identify the actual Credential Issuer when it is using third party service provider as a domain to host its metadata file, in which case, the JWT SHOULD contain `sub` (Subject) claim denoting the third party service provider. If the Wallet supports signed metadata, metadata values conveyed in the signed JWT MUST take precedence over the corresponding values conveyed using plain JSON elements. A `signed_metadata` metadata value SHOULD NOT appear as a claim in the JWT. The Wallet MUST establish trust in the signer of the metadata, and obtain the keys to validate the signature. The concrete mechanism how to do that is out of scope of this specification and may be defined in the profiles of this specification.
```

Comment:
I'm not sure about the security implications of making this feature optional for the wallet. I think if the issuer adds signed metadata, the issuer may want to force the wallet to use this data.

Reply:
How are you suggesting to make it mandatory for the wallet?

Reply:
I am not keen to make this mandatory for the wallet:

1. It can't be implemented without an ecosystem profile that makes clear how to check signatures, and that profile can also make it mandatory to implement for the wallet.
2. If we make it mandatory to implement it then probably becomes a thing we have to conformance test and that makes the conformance testing process more complicated for everyone, even the people that don't need this feature.
3. As an issuer you can essentially force wallets to support signed meta data by publishing all (or at least some critical part) of it's metadata only inside the signed object.
4. It would be inconsistent with https://datatracker.ietf.org/doc/html/rfc8414#section-2.1

Reply:
I'm ok with adding text stating something along the lines of bullet 3.

Reply:
added a suggestion